packages/xo-server/src/api/sr.js

```diff
@@ -523,11 +523,12 @@ export async function probeHba({ host }) {
   forEach(ensureArray(xml.Devlist.BlockDevice), hbaDevice => {
     hbaDevices.push({
       hba: hbaDevice.hba.trim(),
-      id: hbaDevice.id.trim(),
+      // TODO: check if 'id' is a string or number
+      id: typeof hbaDevice.id === 'string' ? hbaDevice.id.trim() : hbaDevice.id,
```

Comment:
To avoid changing API, you can do the following:

```suggestion
      // always consider id as a string even if it looks like a number in the XML
      id: String(hbaDevice.id).trim(),
```